Source/Events.Processing/Filters/FiltersService.cs

```diff
@@ -174,5 +307,107 @@ bool TryStartStreamProcessors(IEnumerable<(TenantId, FilterRegistrationResult<Re
             IReverseCallDispatcher<FiltersClientToRuntimeMessage, FilterRuntimeToClientMessage, FiltersRegistrationRequest, FilterRegistrationResponse, FilterEventRequest, FilterResponse> dispatcher,
             Failure failure,
             CancellationToken cancellationToken) => dispatcher.Reject(new FilterRegistrationResponse { Failure = failure }, cancellationToken);
+
+        bool PartitionedTryStartStreamProcessors(IEnumerable<(TenantId, FilterRegistrationResult<RemoteFilterDefinition>)> registrationResults, out Failure failure)
+        {
+            failure = null;
+            var failedRegistrationReasons = registrationResults
+                                                .Where(tenantAndResult => !tenantAndResult.Item2.Succeeded)
+                                                .Select(tenantAndResult => $"For tenant '{tenantAndResult.Item1}':\n\t{string.Join("\n\t", tenantAndResult.Item2.FailureReason.Value.Split("\n"))}");
+            if (failedRegistrationReasons.Any())
+            {
+                var failureMessage = $"Failed to register filter:\n\t";
+                failureMessage += string.Join("\n\t", failedRegistrationReasons);
+                failure = new Failure(FiltersFailures.FailedToRegisterFilter, failureMessage);
+                return false;
+            }
+
+            foreach ((var tenant, var result) in registrationResults)
+            {
+                _ = result.FilterStreamProcessor.Start();
```

Comment:
What happens with this `Task`? It seems like the lifetime of that thread of execution is handled somewhere else, so I'm not sure it makes sense to return a `Task` on that method?